### PR TITLE
UPS agent & integration test

### PR DESCRIPTION
**Pull request by davidvng**
## Description

Add UPS agent with MIB
Add UPS integration test with SNMP simulation data

## Motivation and Context

Completes requirement for UPS agent

## How Has This Been Tested?

Agent tested on UPS Falcon models at UCSD
Integration test passes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

**Comment by BrianJKoopman** [comment]
In preparation for the review I went ahead and moved the ups agent into the socs package structure, and updated the paths that needed it in the docs/tests/etc. This follows the new structure in place since the merger of #352. Hope to get to the review soon.

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
This looks great! A couple of small requests/comments/questions.

File: socs/agents/ups/agent.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                                           'connected': True}
+            oid_cache['timestamp'] = timestamp
+    # This is a TypeError due to nothing coming back from the yield,
+    # so get_result is None here and can't be iterated.
+    except TypeError:
+        oid_cache['ups_connection'] = {'last_attempt': time.time(),
+                                       'connected': False}
+        raise ConnectionError('No SNMP response. Check your connection.')
+
+    return oid_cache
```
[the remark this answers is not in the document]
Comment (davidvng) [reply socs/agents/ups/agent.py:145]: Yeah, this is definitely doable for the ibootbar and ups agents. The M1000 would need some work because of the timings. I can look into starting this sometime.

File: socs/agents/ups/agent.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+                self.lastGet = time.time()
+
+                # Publish to feed
+                message = _build_message(get_result, read_time)
+                self.log.debug("{msg}", msg=message)
+                session.app.publish_to_feed('ups', message)
+            except ConnectionError as e:
+                self.log.error(f'{e}')
+                yield dsleep(1)
+                self.log.info('Trying to reconnect.')
```
[the remark this answers is not in the document]
Comment (davidvng) [reply socs/agents/ups/agent.py:277]: Currently, the connection dropping is a bit of a mystery. Sometimes it crashes the whole agent and so needs to be initialized, other times `SNMPTwister` handles the reconnection. I think the former happens when the UPS is turned off and on again. Further testing needs to be done, a bit difficult without bothering everything else going on for SAT1 testing.

[the remark this answers is not in the document]
Comment (BrianJKoopman) [reply socs/agents/ups/agent.py:346]: Gotcha, yeah, when we find a good time to test this it'd be good to work out the edge cases. Doesn't have to hold up this PR though.

**Review by BrianJKoopman — APPROVED** [review]
Thanks for the updates! I think this is ready to go.
